### PR TITLE
JAX docs tutorial: Debugging 101

### LINKED ISSUE #18585: Tracking issue: Documentation update
Started with #18269

The plan is to create an entirely new cohesive documentation structure in `docs/tutorials`, to move sections of the existing docs there, and then to remove the old versions leaving HTTP redirects in their place.

Existing drafts (on the main branch) can be seen at https://jax.readthedocs.io/en/latest/tutorials/: note that this is currently unlisted in the website table of contents.
### Indroductory
- [x] Quickstart @jakevdp #18613
- [x] Key Concepts @jakevdp #20347 
- [x] ~Thinking in JAX @8bitmp3 #18581~ removed in #20802
- [x] Just-in-time compilation @jakevdp #18627
- [x] Automatic vectorization @jakevdp #18626
- [x] Automatic differentiation @8bitmp3 #18676
- [x] Debugging @8bitmp3 #18582
- [x] Pseudorandom numbers @jakevdp #18631
- [x] Working with PyTrees @mtthss #18864
- [x] Introduction to sharded computation @jakevdp #20779
- [x] Stateful computations @jakevdp #20732
- [ ] Example: Writing a simple neural network

### Intermediate
- [x] Parallel computation
- [x] Advanced automatic differentiation @8bitmp3 
- [x] Advanced vectorization (collectives, batched map)
- [x] Gradient checkpointing
- [x] Advanced debugging
- [x] External callbacks @jakevdp #18616
- [ ] ~Profiling and performance @dfm~ *punting on this for now*

### Advanced
- [ ] How JAX primitives work
- [ ] Understanding Jaxprs
- [ ] Writing a custom Jaxpr interpreter
- [ ] Ahead-of-time lowering and computation
- [ ] Custom operations for GPUs with C++ and CUDA
- [ ] Pallas: a JAX kernel language

### Reference
- [x] Installation @8bitmp3 #18580 moved to main page in #20784

### PR DESCRIPTION
**Pull request by 8bitmp3**
@jakevdp 

https://github.com/google/jax/issues/18585

### CODE REVIEW

File: docs/tutorials/debugging.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
+# Debugging 101
+
+This tutorial introduces you to a set of built-in JAX debugging methods — `jax.debug.print`, `jax.debug.breakpoint`, and `jax.debug.callback` — that you can use with various JAX transformations.
```
Comment (jakevdp) [inline docs/tutorials/debugging.md:3]: Use cross-referencing syntax when referencing JAX APIs and functions, e.g. ```{func}`jax.debug.print` ```

File: docs/tutorials/debugging.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+Let's begin with `jax.debug.print`.
+
+## JAX `debug.print` for high-level debugging
+
+Python's built-in `print` function can work with some JAX transformations, such as `jax.grad` and `jax.vmap`.
```
Comment (jakevdp) [inline docs/tutorials/debugging.md:9]: not totally true... `print` will print the tracer, which even for `grad` and `vmap` is not always what the user expects.

**Review by jakevdp — COMMENTED** [review]
Looks good overall.

Should we add a discussion of `jax.debug_nans` and `jax.debug_infs`?

File: docs/tutorials/debugging.md [hunk 3]
```diff
@@ -1,9 +1,145 @@
...
+  def false_fn(x):
+    jax.debug.breakpoint()
+  lax.cond(is_finite, true_fn, false_fn, x)
+
+@jax.jit
+def f(x, y):
+  z = x / y
+  breakpoint_if_nonfinite(z)
+  return z
+f(2., 0.) # ==> Pauses during execution
```
Comment (jakevdp) [inline docs/tutorials/debugging.md:120]: I suspect we'll need the `skip-execution` tag here, or else the doc build will timeout

Comment (8bitmp3) [reply docs/tutorials/debugging.md:122]: :+1:

**Review by jakevdp — APPROVED** [review]
